waku/v2/protocol/filter/filter_subscribers.go

```diff
@@ -36,6 +36,17 @@ func (sub *Subscribers) Append(s Subscriber) int {
 	return len(sub.subscribers)
 }
 
+func (subs *Subscribers) SubscriberHasContentTopic(sub Subscriber, topic string) bool {
+	subs.Lock()
+	defer subs.Unlock()
+	for _, filter := range sub.filter.ContentFilters {
```

Comment:
This might be a bit too subtle, the only thing we need from Subscribers is the lock, to serialize properly with RemoveContentFilters. A better option might be to modify `Subscribers.Items()` to take the content topic as a parameter and only push the matching subscribers into the channel.